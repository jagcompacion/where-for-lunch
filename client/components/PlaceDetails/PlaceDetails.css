@import 'styles/variables.css';

.name {
  min-height: 2rem;
  font-size: 2rem;
  margin-bottom: calc($base-spacing * 2);
}
.box {
  min-height: 10rem;
  background-color: white;
  color: $black;
  background: linear-gradient(to bottom, rgba(255,255,255,1) 0%,rgba(255,255,255,0.5) 100%);
  border-radius: 1rem;
  padding: calc($base-spacing * 2);
  > div {
    margin-bottom: calc($base-spacing / 2);
  }
  .ratingWrapper {
    display: flex;
  }
  .rating {
    position: realtive;
    display: inline;
  }
  .ratingScore, .stars {
    display: inline;
  }
  .stars {
    position: relative;
    vertical-align: middle;
    display: inline-block;
    color: $starGrey;
    overflow: hidden;
    margin-left: calc($base-spacing / 2);
  }
  .fullStars {
    position: absolute;
    left: 0;
    top: 0;
    white-space: nowrap;
    overflow: hidden;
    color: $starYellow;
  }
  .emptyStars:before, .fullStars:before {
    content:'\2605\2605\2605\2605\2605';
    font-size: 1rem;
  }
  .photos {
    display: flex;
    margin-left: -10px;
    margin-right: -10px;
    .photo {
      width: 33.33%;
      padding-left: 10px;
      padding-right: 10px;
      img {
        width: 100%;
        object-fit: cover;
        height: 120px;
      }
    }
  }
  .operation {
    margin-left: 10px;
  }
}
