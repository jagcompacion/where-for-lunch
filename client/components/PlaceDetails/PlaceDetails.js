import React from 'react';
import PropTypes from 'prop-types';
import { tConvert, insertStr } from '../../lib/utils';
import styles from './PlaceDetails.css';
import GoogleMap from '../GoogleMap/GoogleMap';

const PlaceDetails = ({ place }) => {
  let regular = null;
  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
  if (place.hours) {
    regular = place.hours.find(item => item.hours_type === 'REGULAR');
  }

  return (
    <div className={styles.root}>
      <div className={styles.name}>{ place.name || 'Where for lunch?' }</div>
      <div className={styles.box}>
        <div className={styles.photos}>
          {
            place.photos && place.photos.map((item, index) => (
              <div key={index} className={styles.photo}>
                <img src={item} />
              </div>
            ))
          }
        </div>
        <div>Address: { place.address }</div>
        <div>Contact no: { place.phone }</div>
        <div>Category: { place.categories && place.categories.join(', ') }</div>
        <div>Price range: { place.price }</div>
        <div className={styles.ratingWrapper}>
          Ratings:{' '}
          { place.rating &&
          <div className={styles.rating}>
            <div className={styles.ratingScore}>{ place.reviewCount } reviews</div>
            <div className={styles.stars}>
              <div className={styles.emptyStars}></div>
              <div className={styles.fullStars} style={{ width: `${place.rating / 5 * 100}%` }}></div>
            </div>
          </div>
          }
        </div>
        <div>Hours of operation:</div>
        {
          regular && regular.open.map((item, index) => (
            <div className={styles.operation} key={index}>{days[item.day]}: {tConvert(insertStr(item.start, '2', ':'))} - {tConvert(insertStr(item.end, '2', ':'))}</div>
          ))
        }
        {place.coordinates &&
          <GoogleMap
            coordinates={{
              lat: place.coordinates.latitude,
              lng: place.coordinates.longitude,
            }}
          />
        }
      </div>
    </div>
  );
};

PlaceDetails.propTypes = {
  place: PropTypes.object,
};

export default PlaceDetails;
