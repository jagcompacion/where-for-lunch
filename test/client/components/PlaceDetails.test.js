import React from 'react';
import { shallow } from 'enzyme';
import PlaceDetails from 'client/components/PlaceDetails/PlaceDetails';
import renderer from 'react-test-renderer';

test('With Enzyme, PlaceDetails component not render rating section when no rating passed over', () => {
  const place = { hehe: 'haha' };
  const wrapper = shallow(
    <PlaceDetails place={place} />,
  );
  const p = wrapper.find('.rating');
  expect(p.length).toBe(0);
});

test('With Enzyme, PlaceDetails component show have 3 photos displayed', () => {
  const place = {
    photos: [
      'http://image1.jpg',
      'http://image2.jpg',
      'http://image2.jpg',
    ],
  };
  const wrapper = shallow(
    <PlaceDetails place={place} />,
  );
  const p = wrapper.find('.photo');
  expect(p.length).toBe(3);
});

test('With Enzyme, PlaceDetails component show have 3 operation hours displayed', () => {
  const place = {
    hours: [
      {
        hours_type: 'REGULAR',
        open: [
          {
            is_overnight: false,
            end: '2200',
            day: 0,
            start: '1730',
          },
          {
            is_overnight: false,
            end: '2200',
            day: 1,
            start: '1730',
          },
          {
            is_overnight: false,
            end: '2200',
            day: 2,
            start: '1730',
          },
        ],
      },
    ],
  };
  const wrapper = shallow(
    <PlaceDetails place={place} />,
  );
  const p = wrapper.find('.operation');
  expect(p.length).toBe(3);
});

test('With Jest snapshot, PlaceDetails component renders rating section when present', () => {
  const place = { hehe: 'haha', rating: 3.5 };
  const placeComponent = renderer
    .create(<PlaceDetails place={place} />)
    .toJSON();
  expect(placeComponent).toMatchSnapshot();
});
