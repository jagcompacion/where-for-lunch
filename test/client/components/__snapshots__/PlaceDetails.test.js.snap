// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, PlaceDetails component renders rating section when present 1`] = `
<div
  className="root"
>
  <div
    className="name"
  >
    Where for lunch?
  </div>
  <div
    className="box"
  >
    <div
      className="photos"
    />
    <div>
      Address: 
    </div>
    <div>
      Contact no: 
    </div>
    <div>
      Category: 
    </div>
    <div>
      Price range: 
    </div>
    <div
      className="ratingWrapper"
    >
      Ratings:
       
      <div
        className="rating"
      >
        <div
          className="ratingScore"
        >
           reviews
        </div>
        <div
          className="stars"
        >
          <div
            className="emptyStars"
          />
          <div
            className="fullStars"
            style={
              Object {
                "width": "70%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div>
      Hours of operation:
    </div>
  </div>
</div>
`;
